paper.md

```diff
@@ -54,282 +54,140 @@ While containers solve a big part of the "dependency hell" problem
 image is not suitable for implementing workflows associated to complex 
 application testing or validating scientific explorations 
 [@zheng_integrating_2015]. For example, a workflow might involve 
-executing two stages, both of them requiring conflicting versions of a 
+executing two steps, both of them requiring conflicting versions of a 
 language runtime (e.g. Python 2.7 **and** Python 3.6). In this 
 scenario, users could resort to solving such conflicts with the use of 
 package managers, but this defeats the purpose of containers, which is 
 to _not_ have to do this sort of thing inside a container. More 
 generally, the more complex a container image definition gets (a more 
 complex `Dockerfile`), the more "monolithic" it gets, and thus the 
 less maintainable and reusable it is. On the other hand, if an 
-experimentation pipeline can be broken down into finer granularity 
+experimentation pipeline can be broken down into finer granular 
 units, we end up having pieces of logic that are easier to maintain 
 and reuse.
 
 ![An end-to-end example of a workflow. On the left we have the 
-`.workflow` file that defines the workflow. On the right, a pictorial 
+`.yml` file that defines the workflow. On the right, a pictorial 
 representation of it.
-](./figures/casestudy.pdf){#fig:casestudy}
+](./figures/casestudy.png){#fig:casestudy}
+
+Through this paper, we propose a protocol in which a
```

Comment:
the term "protocol" in CS contexts refers to networking protocols. Maybe we could find another term. 

In general, I think the introduction section is missing the three components we discussed earlier: 1) what's the problem, 2) what's the current state of the art and 3) what are our contributions. For 1) before diving into containers, we can first describe the issues of reproducibility that scientists face. For 2) we can briefly discuss in general the categories of tools that exist (what we'll expand on the related work section). For 3) we can actually list our contributions explicitly, like "The contributions of this paper are:" and then show the list.